src/couch/src/couch_file.erl

```diff
@@ -338,12 +339,33 @@ init_status_error(ReturnPid, Ref, Error) ->
     ReturnPid ! {Ref, self(), Error},
     ignore.
 
+last_read(Fd) when is_pid(Fd) ->
+    ReadTsSec = case process_info(Fd, dictionary) of
+        {dictionary, Dict} ->
+            case lists:keyfind(read_timestamp, 1, Dict) of
```

Comment:
Something very similar is used in process_info. Would it make sense to create a function:
```
get(Pid, Key, DefaultValue)
```
We also could consider to create this kind of function in `couch_util:get_value_from_pid/3` if you think we are going to use this approach a lot.